samples/rust/rust_thread.rs

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Rust thread APIs sample
+
+#![no_std]
+#![feature(allocator_api, global_asm)]
+#![feature(test)]
```

Comment:
I think you can remove `test` now since the samples were split (it was only used by one).